src/layouts/docs/BaseLayout/Search/index.tsx

```diff
@@ -71,66 +149,100 @@ async function search(query: string) {
 }
 
 export function Search() {
-  const [searchTerm, setSearchTerm] = useState('');
+  const [searchInput, setSearchInput] = useState<string>('');
   const [results, setResults] = useState<Record<string, Result[]>>({});
-  const [isSearching, setIsSearching] = useState(false);
-  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+  const [isLoading, setIsLoading] = useState<boolean>(false); // Waiting for network call?
+  const [isDirty, setIsDirty] = useState<boolean>(false); // Waiting for debounce?
```

Comment:
don't show stale results if the user has typed something else, regardless of debounce state